sources/Launcher/DYNMarginCalculationLauncher.cpp

```diff
@@ -899,6 +923,45 @@ MarginCalculationLauncher::createOutputs(std::map<std::string, std::string>& map
       writeOutputs(itWorst->second);
     }
   }
+
+  auto variationComparison = [](const std::pair<std::string, SimulationResult>& pair1,
+                                const std::pair<std::string, SimulationResult>& pair2) {
+                                  const SimulationResult& scenarioResult1 = pair1.second;
+                                  const SimulationResult& scenarioResult2 = pair2.second;
+                                  return scenarioResult1.getVariation() < scenarioResult2.getVariation();
+                              };
+
+  if (!bestResults.empty()) {
```

Comment:
I'm not sure this works if none of the loadIncrease passed. Why not testing the status on the fly in the loop above?

Reply:
If the load increase pass but every scenarios fails, only the simulation logs for the load increase 0% are in the output. If every load increase fails, no logs are generated at all.

Reply:
In this case, the log of the 0% load increase should be dumped

Reply:
Please store the results on the fly during the main loop above

Reply:
The for loop stocks the scenario with the highest load level which succeeds and stocks the scenario with the lowest load level which fails.

This is not exactly what we want. We want to stock the load increase with the highest load level which succeeds for every scenario. And we want to stock the load increase with the lowest load level which fails for every scenario.

I feel like it may be more tricky to handle that when we are filling the bestResults and worstResults maps with the appropriate scenarios.

Reply:
Yes, but to test the scenarios we first loop over the load increase results. WIth the current solution we parse the load increase results 4 times, whereas we could do it on the fly in the existing loop